modules/oe_content_sub_entity/modules/oe_content_sub_entity_author/src/AuthorSkosUpdater.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_content_sub_entity_author;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\node\NodeInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Updates a given node's 'oe_author' skos field values to sub-entities.
+ *
+ * The created sub-entities will be assigned to 'oe_authors' field.
+ */
+class AuthorSkosUpdater {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Constructs a new AuthorSkosUpdater.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager) {
+    $this->entityTypeManager = $entityTypeManager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * Updates the revisions of a node with new author skos values.
+   *
+   * @param \Drupal\node\NodeInterface $node
+   *   The node.
+   */
+  public function updateNode(NodeInterface $node): void {
+    if (!$node->hasField('oe_author')) {
+      return;
+    }
+
+    /** @var \Drupal\oe_content_sub_entity_author\Entity\AuthorInterface $author */
+    $author = $this->entityTypeManager->getStorage('oe_author')->create([
+      'type' => 'oe_corporate_body',
+    ]);
+
+    $ids = $this->entityTypeManager->getStorage('node')->getQuery()
+      ->allRevisions()
+      ->condition('nid', $node->id())
+      ->sort('vid')
+      ->execute();
+
+    $revisions = $this->entityTypeManager
+      ->getStorage('node')
+      ->loadMultipleRevisions(array_keys($ids));
+
+    /** @var \Drupal\node\NodeInterface $revision */
+    foreach ($revisions as $revision) {
+      // If the field has no value we can skip the creation of new revision on
+      // the sub-entity and leave the 'oe_authors' field empty.
+      if ($revision->get('oe_author')->isEmpty()) {
+        continue;
+      }
+
```

Comment:
Might be add a condition to prevent second migration? In this case update could be run few times if needed.
```
if (!$revision->get('oe_authors')->isEmpty()) {
        continue;
      }
```

Reply:
```
$is_default = $revision->isDefaultRevision();
...
$author->isDefaultRevision($is_default );
```